### PR TITLE
Add X.509 SVID spec

### PR DESCRIPTION
**Pull request by evan2645**
Add X.509 SVID spec
Move SPIFFE spec to `-proposed` to allow for wider review
Update README to remove link to SPIFFE spec. Will re-add when ratified

Following the merge of this PR, I will open another PR which promotes the specs from proposed to ratified. We can use that PR for wider review/discussion of what we've synthesized here.

### CODE REVIEW

File: standards/X509-SVID-proposed.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+TODO
+
+## 1. Introduction
+Perhaps the most important function of SPIFFE is to secure process to process communication. The core standard allows for authentication, however leveraging cryptographic identity to build a secure communication channel is also highly desirable. With TLS being widely adopted, and implemented using X.509-based authentication, the use of X.509 as a SPIFFE SVID is clearly advantageous.
+
+This specification addresses the encoding of SVID information into an X.509 certificate, the constraints which must be set, as well as how to validate X.509 SVIDs.
+
+## 2. SPIFFE ID
+In an X.509 SVID, the corresponding SPIFFE ID is set as a URI type in the Subject Alternative Name extension (SAN extension, see RFC 5280). An X.509 SVID MUST NOT contain more than one URI SAN.
+
```
Comment (y2bishop2y) [inline standards/X509-SVID-proposed.md]: Should we put a link to the section here ? Wondering if anytime we address a section in 5280 we should link to it.

File: standards/X509-SVID-proposed.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+## 3. Hierarchy
+This section discusses the relationship between root, intermediate, and leaf certificates, as well as the requirements placed upon each.
```
Comment (y2bishop2y) [inline standards/X509-SVID-proposed.md]: "intermediates" just want to make sure that the depth can be more than 3 ( root -> inter -> leaf ) 
can be ( root -> interA -> interB -> left ) etc

File: standards/X509-SVID-proposed.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+### 3.1. Leaf Certificates
```
Comment (y2bishop2y) [inline standards/X509-SVID-proposed.md]: I think we should show table of the x509 extensions we are using and the values we want. (Wasn't sure If you moved this somewhere else, had it in the other document )

Comment (y2bishop2y) [reply standards/X509-SVID-proposed.md]: I see that the fields are peppered through out, but a table would help to see the quick view

File: standards/X509-SVID-proposed.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+A signing certificate is itself an SVID. The SPIFFE ID of a signing certificate MUST NOT have a path component, and MUST reside in the trust domain of any leaf SVIDs it issues. A signing certificate MAY be used to issue further signing certificates in the same or different trust domains.
+
+Signing certificates MUST NOT be used for authentication purposes. They serve as validation material only, and may be chained together in typical X.509 fashion, as described in RFC 5280.
+
+## 4. Constraints and Usage
+Leaf and signing certificates carry different X.509 properties - some for security purposes, and some to support their specialized functions. This section describes the constraints and key usage configuration for X.509 SVIDs of both types.
+
+### 4.1. Basic Constraints
+The basic constraints X.509 extension identifies whether the certificate is a signing certificate, as well as the maximum depth of valid certification paths that include this certificate. It is defined in RFC 5280, section 4.2.1.9.
+
```
Comment (y2bishop2y) [inline standards/X509-SVID-proposed.md]: We should use a link to the RFC section also

File: standards/X509-SVID-proposed.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+Valid X.509 SVIDs (both leaf and signing certificates) MUST NOT set the `pathLenConstraint` field. Signing certificates MUST set the `CA` field to `true`, and leaf certificates MUST set the `CA` field to `false`.
+
+### 4.2. Name Constraints
+Name constraints indicate a namespace within which all SPIFFE IDs in subsequent certificates in a certification path MUST be located. They are used to limit the blast radius of a compromised signing certificate, and are defined in RFC 5280, section 4.2.1.10. This section applies to signing certificates only.
```
Comment (y2bishop2y) [inline standards/X509-SVID-proposed.md]: NameConstraints only verify the domain portion of the URI, they don't verify the paths just the Domain segment of the URI

File: standards/X509-SVID-proposed.md (hunk 1 above)
Comment (evan2645) [reply standards/X509-SVID-proposed.md]: Fixed

File: standards/X509-SVID-proposed.md (hunk 2 above)
Comment (evan2645) [reply standards/X509-SVID-proposed.md]: I re-organized it so "intermediate certificates" is at the end... is that more clear?

FWIW, chaining of signing certificates is explicitly called out in 3.2.

File: standards/X509-SVID-proposed.md (hunk 3 above)
Comment (evan2645) [reply standards/X509-SVID-proposed.md]: I added an Appendix section with a reference table

File: standards/X509-SVID-proposed.md (hunk 4 above)
Comment (evan2645) [reply standards/X509-SVID-proposed.md]: Fixed

File: standards/X509-SVID-proposed.md (hunk 5 above)
Comment (evan2645) [reply standards/X509-SVID-proposed.md]: I added a blurb to clarify that it applies to the trust domain and not the path

**Comment by y2bishop2y** [comment]
LGTM

**Pull request merged**